os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -149,20 +150,52 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 	/* Allocate memory to hold the ELF image */
 
 #ifdef CONFIG_APP_BINARY_SEPARATION
-#ifdef CONFIG_DEBUG_MM_HEAPINFO
-	ARCH_GET_RET_ADDRESS
-	loadinfo->textalloc = (uintptr_t)mm_malloc(loadinfo->uheap, textsize + datasize, retaddr);
-#else
-	loadinfo->textalloc = (uintptr_t)mm_malloc(loadinfo->uheap, textsize + datasize);
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD
+	uint32_t rosize = loadinfo->rosize;
+	loadinfo->rosize += loadinfo->datasize;
+#endif
+	loadinfo->textsize = 1 << mpu_log2regionceil(0, loadinfo->textsize);
```

Comment:
There are different prototype between ARM-M and ARM-R about `mpu_log2regionceil` .
Please add condition.

Reply:
Done.